internal/services/storage/migration/account.go

```diff
@@ -204,3 +211,810 @@ func accountSchemaForV0AndV1() map[string]*pluginsdk.Schema {
+						Optional: true,
+						Default:  false,
+					},
+
+					"choice": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(storage.MicrosoftRouting),
+							string(storage.InternetRouting),
+						}, false),
+						Default: string(storage.MicrosoftRouting),
+					},
+				},
+			},
+		},
+
+		"share_properties": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"cors_rule": helpers.SchemaStorageAccountCorsRule(true),
+
+					"retention_policy": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"days": {
+									Type:         pluginsdk.TypeInt,
+									Optional:     true,
+									Default:      7,
+									ValidateFunc: validation.IntBetween(1, 365),
+								},
+							},
+						},
+					},
+
+					"smb": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"versions": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice([]string{
+											"SMB2.1",
+											"SMB3.0",
+											"SMB3.1.1",
+										}, false),
+									},
+								},
+
+								"authentication_types": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice([]string{
+											"NTLMv2",
+											"Kerberos",
+										}, false),
+									},
+								},
+
+								"kerberos_ticket_encryption_type": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice([]string{
+											"RC4-HMAC",
+											"AES-256",
+										}, false),
+									},
+								},
+
+								"channel_encryption_type": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type: pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice([]string{
+											"AES-128-CCM",
+											"AES-128-GCM",
+											"AES-256-GCM",
+										}, false),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		//lintignore:XS003
+		"static_website": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"index_document": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
fwiw you don't need these for a state migration - only the Type and Required/Optional/Computed iirc

```suggestion
```